packages/x-data-grid/src/DataGrid/useDataGridProps.ts

```diff
@@ -58,6 +58,7 @@ export const DATA_GRID_PROPS_DEFAULT_VALUES: DataGridPropsWithDefaultValues = {
   hideFooterSelectedRowCount: false,
   ignoreDiacritics: false,
   ignoreValueFormatterDuringExport: false,
+  indeterminateCheckboxBehavior: 'select',
```

Comment:
This is a change from the current behavior.
Any disagreements on this being a non breaking change?

Reply:
This is clearly not my decision to take
I would wait for v8 since it's not that far away (alpha is in October I think)

Reply:
I know we already had a naming suggestions thread, but I am throwing in one more to consider
`indeterminateCheckboxAction`

Reply:
Thanks @arminmeh, this indeed looks like a better name 👍